sa/proto/sa.proto

```diff
@@ -39,7 +39,9 @@ service StorageAuthority {
         rpc DeactivateRegistration(RegistrationID) returns (core.Empty) {}
         rpc DeactivateAuthorization(AuthorizationID) returns (core.Empty) {}
         rpc NewOrder(core.Order) returns (core.Order) {}
+        rpc UpdateOrder(core.Order) returns (core.Order) {}
```

Comment:
Sure. That makes sense. AFAICT there shouldn't be a need to change an order after creation except for the status & certificate serial.

Reply:
@jsha So something like
`rpc FinalizeOrder(core.Order) returns (core.Order) {}`

Where the SA's implementation only pulled out the `CertificateSerial` field of the order provided as an argument & then created a model that had that field + `status(final)` and used it for an update?

Or would you rather something like:
`rpc FinalizeOrder(orderID, certificateSerial) returns (core.Order) {}` to make that limitation more explicit?

Reply:
I implemented the former to keep this PR moving.

Reply:
> rpc FinalizeOrder(core.Order) returns (core.Order) {}
> Where the SA's implementation only pulled out the CertificateSerial field of the order provided as an argument & then created a model that had that field + status(final) and used it for an update?

This is fine, though I'd recommend an explicit query `UPDATE orders SET status = ? WHERE id = ? AND status = "pending"` rather than using the whole Order model in this case.

Reply:
Sounds good. I'll make that change shortly.

Reply:
Done in 9496ef5